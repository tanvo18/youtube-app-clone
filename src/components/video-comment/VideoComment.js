import React, { Component } from 'react';
import VideoCommentStyled from './VideoCommentStyled';
import PropTypes from 'prop-types';

class VideoComment extends Component {
  render() {

    return (
      <VideoCommentStyled>
        <div className="avatar-wrap">
          <img src={this.props.avatar} alt="avatar"/>
        </div>
        <div className="comment-wrap">
          <p>{this.props.username}</p>
          <p className="user-comment">{this.props.userComment}</p>
        </div>
      </VideoCommentStyled>
    );
  }
}

VideoComment.defaultProps = {
  avatar: '',
  username: '',
  userComment: '',
};

VideoComment.propTypes = {
  avatar: PropTypes.string,
  username: PropTypes.string,
  userComment: PropTypes.string,
};

export default VideoComment;
