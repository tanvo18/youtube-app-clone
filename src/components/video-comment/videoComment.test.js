import React, { Component } from 'react';
import VideoComment from './VideoComment';
import renderer from 'react-test-renderer';

const props = {
  avatar: 'https://yt3.ggpht.com/-6Kja_qUGQIU/AAAAAAAAAAI/AAAAAAAAAAA/rEUlDi7aSTI/s28-c-k-no-mo-rj-c0xffffff/photo.jpg',
  username: 'Vibha Upadhyay',
  userComment: 'It is fun'
};

describe('test videoComment', () => {
  it('test user comment text', () => {
    const wrapper = shallow(<VideoComment {...props} />);
    expect(wrapper.find('.user-comment').text()).toBe('It is fun');
  });

  it('should render correctly', () => {
    const tree = renderer
      .create(<VideoComment {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});