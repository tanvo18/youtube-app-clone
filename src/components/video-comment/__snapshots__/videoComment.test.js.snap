// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test videoComment should render correctly 1`] = `
<li
  className="sc-bdVaJa iuEGl"
>
  <div
    className="avatar-wrap"
  >
    <img
      alt="avatar"
      src="https://yt3.ggpht.com/-6Kja_qUGQIU/AAAAAAAAAAI/AAAAAAAAAAA/rEUlDi7aSTI/s28-c-k-no-mo-rj-c0xffffff/photo.jpg"
    />
  </div>
  <div
    className="comment-wrap"
  >
    <p>
      Vibha Upadhyay
    </p>
    <p
      className="user-comment"
    >
      It is fun
    </p>
  </div>
</li>
`;
